include/psen_scan/start_request.h

```diff
@@ -0,0 +1,137 @@
+
+#include <psen_scan/scanner_configuration.h>
+#include <psen_scan/crc.h>
+
+#pragma pack(push, 1)
+
+namespace psen_scan
+{
+
+/**
+ * @brief Frame containing all necessary fields for a Start Request.
+ *
+ * Unless otherwise indicated the byte order is little endian.
+ *
+ */
+typedef struct StartRequest
+{
+public:
+  uint32_t crc_;                  /**< A CRC32 of all the following fields. */
+  uint32_t seq_number_;
+  uint64_t const RESERVED_ { 0 };       /**< Use all zeros */
+  uint32_t const OPCODE_ { htole32(0x35) };         /**< Constant 0x35. */
+  uint32_t target_ip_;            /**< Byte order: big endian */
+  uint16_t target_udp_port_;  /**< Byte order: big endian */
+
+  /**< The following 'enable' fields are a 1-byte mask each.
+   * Only the last 4 bits (little endian) are used, each of which represents a device.
+   * For example, (1000) only enables the Master device, while (1010) enables both the Master
+   * and the second Slave device.
+   */
+  uint8_t device_enabled_;
+  uint8_t intensity_enabled_;
+  uint8_t point_in_safety_enabled_;
+  uint8_t active_zone_set_enabled_;
+  uint8_t io_pin_enabled_;
+  uint8_t scan_counter_enabled_;
+  uint8_t speed_encoder_enabled_;    /**< 0000000bin disabled, 00001111bin enabled.*/
+  uint8_t diagnostics_enabled_;
+
+  /**< The following 'angle' and 'resolution' fields are all in tenth of degrees */
+  uint16_t master_start_angle_;
+  uint16_t master_end_angle_;
+  uint16_t master_resolution_;
+  uint16_t slave1_start_angle_;
+  uint16_t slave1_end_angle_;
+  uint16_t slave1_resolution_;
+  uint16_t slave2_start_angle_;
+  uint16_t slave2_end_angle_;
+  uint16_t slave2_resolution_;
+  uint16_t slave3_start_angle_;
+  uint16_t slave3_end_angle_;
+  uint16_t slave3_resolution_;
+
+public:
+  StartRequest(const ScannerConfiguration& scanner_configuration)
+  : target_udp_port_(htobe16(scanner_configuration.target_udp_port_))
+  , master_start_angle_(htole16(scanner_configuration.master_start_angle_))
+  , master_end_angle_(htole16(scanner_configuration.master_end_angle_))
+  , master_resolution_(htole16(scanner_configuration.master_resolution_))
+  , slave1_start_angle_(htole16(scanner_configuration.slave1_start_angle_))
+  , slave1_end_angle_(htole16(scanner_configuration.slave1_end_angle_))
+  , slave1_resolution_(htole16(scanner_configuration.slave1_resolution_))
+  , slave2_start_angle_(htole16(scanner_configuration.slave2_start_angle_))
+  , slave2_end_angle_(htole16(scanner_configuration.slave2_end_angle_))
+  , slave2_resolution_(htole16(scanner_configuration.slave2_resolution_))
+  , slave3_start_angle_(htole16(scanner_configuration.slave3_start_angle_))
+  , slave3_end_angle_(htole16(scanner_configuration.slave3_end_angle_))
+  , slave3_resolution_(htole16(scanner_configuration.slave3_resolution_))
+  {
+    // TODO: What to do with seq_number???
+    setTargetIP(scanner_configuration.target_ip_);
+    setEnableFields(scanner_configuration);
+    setCRC();
+  }
```

Comment:
For easier readability of the interface, I suggest to split declaration and definition of the functions.

Reply:
done